pkl-core/src/test/kotlin/org/pkl/core/parser/ParserComparisonTestInterface.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © 2024-2025 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.core.parser
+
+import java.nio.file.Path
+import kotlin.io.path.pathString
+import kotlin.io.path.readText
+import org.antlr.v4.runtime.ANTLRInputStream
+import org.antlr.v4.runtime.CommonTokenStream
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.SoftAssertions
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.parallel.Execution
+import org.junit.jupiter.api.parallel.ExecutionMode
+import org.pkl.core.parser.antlr.PklLexer
+import org.pkl.core.parser.antlr.PklParser
+
+@Execution(ExecutionMode.CONCURRENT)
+interface ParserComparisonTestInterface {
+  @Test
+  @Execution(ExecutionMode.CONCURRENT)
+  fun compareSnippetTests() {
```

Comment:
Nit: this would be a good candidate for `@ParameterizedTest`.

But, this is temporary code anyways, so, it's not a big deal.